src/integration_tests/test_with_instance.py

```diff
@@ -0,0 +1,128 @@
+import os, json
+from fabric.api import settings
+from tests import base
+from buildercore import bootstrap, cfngen, lifecycle, utils, core
+from buildercore.config import BOOTSTRAP_USER
+import cfn
+import logging
+
+logging.disable(logging.NOTSET) # re-enables logging during integration testing
+
+# the intent here is:
+# * launch an ec2 instance without running highstate
+# * run any logic against it that needs a running ec2 instance
+# * optionally leave the instance running while debugging happens
+
+class One(base.BaseCase):
+    @classmethod
+    def setUpClass(self): # cls, not self
+        super(One, self).setUpClass()
+
+        # to re-use an existing stack, ensure self.reuse_existing_stack is True
+        # this will read the instance name from a temporary file (if it exists) and
+        # look for that, creating it if doesn't exist yet
+        # also ensure self.cleanup is False so the instance isn't destroyed after tests complete
+        self.reuse_existing_stack = False
```

Comment:
if this read an environment variable it could be set for local usage without changing the code